src/qonnx/custom_op/ChannelsLast/wrapped_ops.py

```diff
@@ -0,0 +1,488 @@
+            outputs=[self.onnx_node.output[0]],
+            value=helper.make_tensor(
+                name="const_tensor",
+                data_type=TensorProto.FLOAT,
+                dims=values.shape,
+                vals=values.flatten().astype(float),
+            ),
+            name=self.onnx_node.name,
+        )
+
+    def verify_node(self):
+        node = self.onnx_node
+
+        verification_successful = True
+        info_messages = []
+
+        wrapper_info = ChannelsLastWrappedOp.verify_node(self)
+        info_messages.extend(wrapper_info)
+
+        # verify number of attributes
+        num_of_attr = 5
+        if len(node.attribute) == num_of_attr:
+            info_messages.append("The number of attributes is correct")
+        else:
+            info_messages.append(
+                """The number of attributes is incorrect,
+            {} should have {} attributes""".format(
+                    node.op_type, num_of_attr
+                )
+            )
+            verification_successful = False
+
+        # verify that all necessary attributes exist
+        try:
+            self.get_nodeattr("dilations")
+            self.get_nodeattr("group")
+            self.get_nodeattr("kernel_shape")
+            self.get_nodeattr("pads")
+            self.get_nodeattr("strides")
+            info_messages.append("All necessary attributes exist")
+        except Exception:
+            info_messages.append(
+                """The necessary attributes do not exist.
+                Conv needs the following attributes:
+                dilations, group, kernel_shape, pads, strides"""
+            )
+            verification_successful = False
+
+        # verify that attributes have the correct datatype.
+        try:
+            assert isinstance(self.get_nodeattr("kernel_shape"), RepeatedScalarContainer)
+            assert isinstance(self.get_nodeattr("pads"), RepeatedScalarContainer)
+            assert isinstance(self.get_nodeattr("strides"), RepeatedScalarContainer)
+            assert isinstance(self.get_nodeattr("dilations"), RepeatedScalarContainer)
+            assert isinstance(self.get_nodeattr("group"), int)
+            info_messages.append("All attributes are of the correct type")
+        except Exception:
+            info_messages.append("One or more attributes are of the wrong datatype")
+            verification_successful = False
+
+        # verify the number of inputs
+        if len(node.input) == 2:
+            info_messages.append("The number of inputs is correct")
+        else:
+            info_messages.append("{} needs 2 data inputs".format(node.op_type))
+            verification_successful = False
+
+        if not verification_successful:
+            raise RuntimeError(
+                f"Verification of node {node.name} failed, please check the " f"attached info messages: {info_messages}"
+            )
+
+        return info_messages
+
+
+class MaxPool(ChannelsLastWrappedOp):
```

Comment:
suggest to move to own module, see comments about Conv

Reply:
Done